syft/pkg/cataloger/java/cataloger.go

```diff
@@ -9,17 +9,21 @@ import (
 
 // NewJavaCataloger returns a new Java archive cataloger object.
 func NewJavaCataloger(cfg Config) *generic.Cataloger {
+	gap := newGenericArchiveParserAdapter(cfg)
```

Comment:
nit: not all of these need to be constructed based on the below conditions

Reply:
moved these down into their respective configuration blocks

Reply:
moved separate adapters into their respective config locations